src/tools/cephfs/cephfs_shell.py

```diff
@@ -0,0 +1,320 @@
+        except AttributeError:
+            return None
+        doc_lines = getattr(self, "do_" + command).__doc__.expandtabs().splitlines()
+        if '' in doc_lines:
+            blank_idx = doc_lines.index('')
+            usage = doc_lines[:blank_idx]
+            description = doc_lines[blank_idx + 1:]
+        else:
+            usage = doc_lines
+            description = []
+        parser = argparse.ArgumentParser(
+            prog=command,
+            usage='\n'.join(usage),
+            description='\n'.join(description),
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+        for args, kwargs in argparse_args:
+            parser.add_argument(*args, **kwargs)
+        return parser
+
+    def onecmd(self, line):
+        """Global error catcher"""
+        try:
+            res = cmd.Cmd.onecmd(self, line)
+            if self.interactive: 
+                self.set_prompt()
+            return res
+        except ConnectionError as e:
+            print("***", e)
+        except KeyboardInterrupt:
+            print("Command aborted")
+        except Exception as e:
+            print(e)
+            traceback.print_exc(file=sys.stdout)
+
+
+    def do_mkdir(self,arglist):
+        """Create directory.
+           Usage: mkdir [-m MODE] <dir_name> """    
+        try:
+            self.parser = argparse.ArgumentParser(description='Create Directory.')
+            self.parser.add_argument('dir_name', type = str, help = ' Name of new_directory.')
+            self.parser.add_argument('-m', '--mode', action='store', help='Sets the access mode for the new directory.',type=str)
+            args = self.parser.parse_args(shlex.split(arglist))
+            path = bytes('/' + args.dir_name, encoding="ascii")
+            if args.mode :
+                permission = int(args.mode,8)
+            else :
+                permission = 0o777
+            print(path.decode("utf-8"),permission)
+            cephfs.mkdirs(path, permission)
+        except:
+             traceback.print_exc(file=sys.stdout)
+       
+    def do_put(self, arglist):
+        """ Copy a file to Ceph File System from Local Directory.
+            Usage : put <local_path> <remote_path> """
+        try:
+            self.parser = argparse.ArgumentParser(description='Copy a file to Ceph File System from Local Directory.')
+            self.parser.add_argument('local_path', type = str, help = 'Path of the file in the local system')
+            self.parser.add_argument('remote_path', type = str,help = 'Path of the file in the remote system')
+            args = self.parser.parse_args(shlex.split(arglist))
+            copy_from_local(args.local_path, args.remote_path)
+        except :
+             traceback.print_exc(file=sys.stdout)
+        
+
+    def do_get(self, arglist):
+        """ Copy a file from Ceph File System to Local Directory.
+            Usage : get <local_path> <remote_path> """
```

Comment:
I think 'get' should also be able to write to stdout

Reply:
See do_get() in https://github.com/ceph/ceph/blob/master/src/tools/rados/rados.cc